update documentation

kdembler opened the pull request:
### Summary of changes:
- Updated README.md
- Updated technical styleguide
- Updated technical overview
- Created CONTRIBUTING.md

### Left to do:

- [ ] Create some custom banner for Atlas to be displayed on top of README.md
- [ ] Populate the screenshots section

kdembler committed:
update documentation

drillprop commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,33 @@
+
+### Contributing TL;DR
+
+1. Always make sure to create an issue/comment on an existing issue before you start any work.
```
```suggestion
1. Always make sure to create an issue or comment on an existing issue before you start any work.
```
A small thing, but I don't like `/` here, this could suggest that you can `create an issue on an existing issue` :D I mean I read that first time, had to go back and read again to understand the whole sentence.

drillprop commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,33 @@
+If you want to contribute, the very first place to stop at is the [issues list](https://github.com/Joystream/atlas/issues). It's a very good practice to base any work on an issue. If you want to add some functionality to the app, create an issue with your proposal first, so we can discuss. If it turns out the team disagrees with the change you're proposing, you will not waste time doing wasted wo ...
+
+Once you know what you will be working on and created/commented on an appropriate issue, it's time to get to work. Before you begin tough, please fork the main repo, so you have your own copy which you can use for development. We work in a forking model, so nobody pushes to the main repo directly - all contributors have their own forks from which they open PRs to the main repo.
+
```
Not sure, but maybe we could add here a link to this video guide for newcomers? https://egghead.io/courses/how-to-contribute-to-an-open-source-project-on-github

drillprop commented on docs/styleguide.md:
```diff
@@ -1,78 +1,161 @@
+3. Event handlers
+4. Conditional returns
+
+Here is an example of a nicely written component:
```
I think we could also add that we prefer using `types` over `interfaces`

kdembler replied:
Added typescript section 👍

toiletgranny committed:
Update README.md

Added a header (based on the convention established in other JS repos), sample app screenshots, and the "Design" part of the readme file.

toiletgranny reviewed (dismissed):
Hey @kdembler, I updated the README.md file with some design goodies.

kdembler committed:
fix lint and tweaks

kdembler committed:
review followup, add section on Typescript

rafalpawlow commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,33 @@
+
+### Contributing TL;DR
+
+1. Always make sure to create an issue or comment on an existing issue before you start any work.
```
Period at the end of a sentence, not present in other bullets

rafalpawlow commented on README.md:
```diff
@@ -1,21 +1,72 @@
+  <a alt="Contributors" href="https://github.com/Joystream/atlas/graphs/contributors">
+    <img alt="Contributors badge" src="https://img.shields.io/github/contributors/joystream/atlas?label=%F0%9F%91%A5%20contributors">
+  </a>
+  <a alt="Styleguide" href="https://github.com/Joystream/atlas/graphs/contributors">
```
This link seems to be wrong

kdembler replied:
oof, good catch

rafalpawlow commented on docs/community/featured-content.md:
```diff
@@ -26,8 +26,9 @@ In general the process for updating the featured content will look as follows:
+- This means the first featured video (with an accompanying video file) will take the first spot in videos that show a preview.
+- For videos submitted with only an ID (and no video file) they will occupy the section below in the order submitted.
 
 IMPORTANT NOTE 2: Anytime a mutation is submitted, it will override whatever was previously set. So if you want to add/remove featured videos from categories--you must list all the ones you do want listed and remove those that you do not want listed anymore.
```
Is the double dash intentional?

kdembler replied:
Don't think so, this was added by a community member

rafalpawlow commented on docs/overview.md:
```diff
@@ -1,44 +1,38 @@
 - Typescript
-- GraphQL (+ [Relay-style pagination](https://graphql.org/learn/pagination/))
-- ESLint, Prettier - for enforcing common code style
+- GraphQL (+ [Relay-style pagination](https://graphql.org/learn/pagination/)) with [Apollo Client](https://www.apollographql.com/docs/react/)
```
Maybe we could also mention Zustand here?

kdembler replied:
Mentioned

rafalpawlow reviewed (changes requested):
I think we are missing info about working with figma icons. Current flow with getting them from API may confuse people.

kdembler committed:
more fixes

kdembler committed:
add section on icons & illustrations

eldiegod commented on docs/community/featured-content.md:
```diff
@@ -26,48 +26,49 @@ In general the process for updating the featured content will look as follows:
 5. Changes are live in Atlas
 
 IMPORTANT NOTE 1: for `category featured videos` the order of these will be the same as taken as the mutation submitted to Orion.
```
What about:

For `category featured videos`, the order of these will be the same as the mutation submitted to Orion.

kdembler replied:
This wasn't part of this PR but updated

eldiegod commented on docs/styleguide.md:
```diff
@@ -1,78 +1,179 @@
+  - `index.ts` - re-exporting anything needed
+  - `Component.tsx` - component code
+  - _(optional)_ `Component.styles.ts` - any styles needed for the component
+  - _(optional)_ `Component.stories.tsx` - Storybook stories exploring different use cases/variants of the component
```
Maybe we should add an extra line here for a unit test?
eg:
- _(optional)_ `Component.test.tsx` - Component's unit tests

kdembler replied:
Let's do that once we actually introduce unit tests

eldiegod reviewed (approved):
Looks good overall, added some notes

kdembler committed:
followup
